connexion/utils.py

```diff
@@ -28,14 +31,23 @@
 }
 
 
-def flaskify_endpoint(identifier):
+def flaskify_endpoint(identifier, randomize=None):
     """
     Converts the provided identifier in a valid flask endpoint name
 
     :type identifier: str
+    :param randomize: If specified, add this many random characters (upper case
+        and digits) to the endpoint name, separated by a pipe character.
+    :type randomize: int | None
     :rtype: str
     """
-    return identifier.replace('.', '_')
+    result = identifier.replace('.', '_')
+    if randomize is None:
+        return result
+
+    chars = string.ascii_uppercase + string.digits
+    return result + '|' + ''.join(
```

Comment:
Since you changed, maybe then move this concatenation (` + '|' + `) to the `''.join(...` ?